cloud-agnostic/core/src/NamedDependency.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { Container, interfaces } from "inversify";
+
+import { Dependency } from "./Dependency";
+import { DependencyConfig } from "./DependencyConfig";
+import { ConfigError } from "./internal";
+
+export abstract class NamedDependency extends Dependency {
+  protected registerNamedInstance<TInstance>(
```

Comment:
This assumes that there is only one class that will be resolved from the parent container and all the other classes in child container are there only to resolve it. That is the case for StorageDependency, but not this cloud agnostic framework in general. There could be multiple classes that you might want to resolve and they should share the same child container.

Reply:
Fixed